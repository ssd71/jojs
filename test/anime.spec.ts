import Anime = require('../src/anime');

const propNames: Array<string> = [
  'aid',
  'eps',
  'count',
  'special_count',
  'rating',
  'votes',
  'tmprating',
  'tmpvotes',
  'review_avg_Rating',
  'reviews',
  'year',
  'type',
  'romaji',
  'kanji',
  'english',
  'other',
  'short_names',
  'synonyms',
  'cat_list',
];

let ins: Anime;

describe('Anime class', () => {
  test('creation of new Anime Objects', () => {
    // Use example anime to init a new Anime object
    // eslint-disable-next-line max-len
    ins = new Anime('1|13|13|3|817|4526|822|4556|870|12|1999-1999|TV Series|Seikai no Monshou|?????|Crest of the Stars|?????\'Crest of the Stars|CotS\'SnM|Hv?zdn? erb\'?????|Space,Plot Continuity,Future,Space Travel,Sci-Fi,Other Planet,Action,Genetic Modification,Novel,War,Shipboard,Slow When It Comes to Love,Military,Romance,Human Enhancement,Large Breasts,Gunfights,Adventure,Nudity,Small Breasts|915321600|922579200|16|0|224618.jpg'.split('|'));
    expect(ins).toBeTruthy();
  });

  test('getters for all properties', () => {
    // Test getters to return example data
    for (let i = 0; i < propNames.length; i++) {
      const res = ins.get(propNames[i]);
      expect(res).toMatchSnapshot();
    }
  });
});
