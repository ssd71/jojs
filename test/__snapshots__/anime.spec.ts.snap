// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime class getters for all properties 1`] = `"1"`;

exports[`Anime class getters for all properties 2`] = `"13"`;

exports[`Anime class getters for all properties 3`] = `"13"`;

exports[`Anime class getters for all properties 4`] = `"3"`;

exports[`Anime class getters for all properties 5`] = `"817"`;

exports[`Anime class getters for all properties 6`] = `"4526"`;

exports[`Anime class getters for all properties 7`] = `"822"`;

exports[`Anime class getters for all properties 8`] = `"4556"`;

exports[`Anime class getters for all properties 9`] = `"870"`;

exports[`Anime class getters for all properties 10`] = `"12"`;

exports[`Anime class getters for all properties 11`] = `"1999-1999"`;

exports[`Anime class getters for all properties 12`] = `"TV Series"`;

exports[`Anime class getters for all properties 13`] = `"Seikai no Monshou"`;

exports[`Anime class getters for all properties 14`] = `"?????"`;

exports[`Anime class getters for all properties 15`] = `"Crest of the Stars"`;

exports[`Anime class getters for all properties 16`] = `"?????'Crest of the Stars"`;

exports[`Anime class getters for all properties 17`] = `"CotS'SnM"`;

exports[`Anime class getters for all properties 18`] = `"Hv?zdn? erb'?????"`;

exports[`Anime class getters for all properties 19`] = `"Space,Plot Continuity,Future,Space Travel,Sci-Fi,Other Planet,Action,Genetic Modification,Novel,War,Shipboard,Slow When It Comes to Love,Military,Romance,Human Enhancement,Large Breasts,Gunfights,Adventure,Nudity,Small Breasts"`;
